addons/web/static/src/core/datetime/datetime_picker_popover.xml

```diff
@@ -14,6 +14,17 @@
                     </button>
                 </t>
             </t>
+            <t t-set-slot="cancel">
+                <t t-if="props.pickerProps.type === 'datetime' or Array.isArray(props.pickerProps.value)">
```

Comment:
as this condition is duplicated, I also suggest to introduce a getter in the component (e.g. `isDateTimeRange`)